dynamodb-stream-reader/src/main/java/no/unit/nva/dynamodb/DynamoDBEventTransformer.java

```diff
@@ -0,0 +1,114 @@
+package no.unit.nva.dynamodb;
+
+import com.amazonaws.services.lambda.runtime.events.models.dynamodb.AttributeValue;
+import elasticsearch.ElasticSearchIndexDocument;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static no.unit.nva.dynamodb.DynamoDBStreamHandler.CONTRIBUTORS_IDENTITY_NAME;
+import static no.unit.nva.dynamodb.DynamoDBStreamHandler.DATE_YEAR;
+import static no.unit.nva.dynamodb.DynamoDBStreamHandler.DESCRIPTION_MAIN_TITLE;
+import static no.unit.nva.dynamodb.DynamoDBStreamHandler.PUBLICATION_TYPE;
+
+public class DynamoDBEventTransformer {
```

Comment:
I am really sceptical about this class because it has no tests and I am not sure whether it is actually necessary.
We can at least consider refactoring it a bit. I have grouped some operations together so that the code is more readable:

suggestion:
```
 public DynamoDBEventTransformer() {
    }

    /**
     * Transforms a nested valuemap read from DynamoDB streamrecord into ElasticSearchIndexDocument.
     *
     * @param identifier of the original dynamoDB record
     * @param valueMap   Map containing the values associated with the record
     * @return A document usable for indexing in elasticsearch
     */
    public ElasticSearchIndexDocument parseValueMap(String elasticSearchIndexName,
                                                    String targetServiceUrl,
                                                    String identifier,
                                                    Map<String, AttributeValue> valueMap) {
        ElasticSearchIndexDocument document =
                new ElasticSearchIndexDocument(elasticSearchIndexName, targetServiceUrl, identifier);
        parseMapValue(document, EMPTY_STRING, valueMap);
        return document;
    }

    private void parseMapValue(ElasticSearchIndexDocument document, String prefix, Map<String, AttributeValue> valueMap) {
        valueMap.forEach((k, v) -> processKeyValueEntry(document, prefix, k, v));
    }

    private void processKeyValueEntry(ElasticSearchIndexDocument document, String prefix, String origninalKey,
                                      AttributeValue value) {
        if (value != null) {
            String newKey = addIndexPrefix(prefix, origninalKey);
            if (isASimpleValue(value)) {
                processSimpleValue(document, value, newKey);
            } else if (valueIsMapElement(value)) {
                parseMapValue(document, newKey, valueAsMap(value));
            } else if (valueIsListElement(value)) {
                parseListValue(document, value, newKey);
            }
        }
    }

    private void parseListValue(ElasticSearchIndexDocument document, AttributeValue value, String newKey) {
        List<AttributeValue> listElements = valueAsList(value);
        for (AttributeValue attributeValue : listElements) {
            processListElement(document, value, newKey, attributeValue);
        }
    }

    private void processListElement(ElasticSearchIndexDocument document, AttributeValue value, String newKey, AttributeValue attributeValue) {
        if (isASimpleValue(attributeValue)) {
            processSimpleValue(document, value, newKey);
        } else {
            parseMapValue(document, newKey, valueAsMap(attributeValue));
        }
    }

    private List<AttributeValue> valueAsList(AttributeValue value) {
        return value.getL();
    }

    private Map<String, AttributeValue> valueAsMap(AttributeValue value) {
        return value.getM();
    }


    private boolean valueIsListElement(AttributeValue v) {
        return valueAsMap(v) == null && valueAsList(v) != null;
    }


    private void processSimpleValue(ElasticSearchIndexDocument document, AttributeValue v, String newKey) {
        if (valueShouldBeIndexed(newKey)) {
            assignValueToIndexDocument(document, newKey, v.getS());
        }
    }

    private boolean valueIsMapElement(AttributeValue v) {
        return valueAsList(v) == null && valueAsMap(v) != null;
    }

    private boolean valueShouldBeIndexed(String newKey) {
        return wantedIndexes.contains(newKey);
    }

    private void assignValueToIndexDocument(ElasticSearchIndexDocument document, String valueKey, String value) {
        switch (valueKey) {
            case PUBLICATION_TYPE:
                document.setResourceType(value);
                break;
            case CONTRIBUTORS_IDENTITY_NAME:
                document.addContributorName(value);
                break;
            case DESCRIPTION_MAIN_TITLE:
                document.setTitle(value);
                break;
            case DATE_YEAR:
                document.setDate(value);
                break;
            default:
                logger.debug(UNKNOWN_VALUE_KEY_MESSAGE, valueKey);
                break;
        }
    }

    private boolean isASimpleValue(AttributeValue attributeValue) {
        return valueAsMap(attributeValue) == null && valueAsList(attributeValue) == null;
    }

    private String addIndexPrefix(String prefix, String k) {
        if (prefix != null && !prefix.isEmpty()) {
            return prefix + SIMPLE_DOT_SEPARATOR + k;
        } else {
            return k;
        }
    }

```